fleetconfig-controller/api/v1alpha1/fleetconfig_types.go

```diff
@@ -306,6 +307,21 @@ type Spoke struct {
 	// ClusterARN is the ARN of the spoke cluster.
 	// This field is optionally used for AWS IRSA registration authentication.
 	ClusterARN string `json:"clusterARN,omitempty"`
+
+	// AddOns are the add-ons to enable for the spoke cluster.
+	AddOns []SpokeAddOn `json:"addOns,omitempty"`
+}
+
+// SpokeAddOn is the configuration for enabling add-on installation on the spoke cluster.
+type SpokeAddOn struct {
```

Comment:
is there the concept of a hubAddOn? if not, maybe we should just name this struct AddOn.